### PR TITLE
Size, addSelect and memberOf DQL functions

### PR DESCRIPTION
**Pull request by patrickvale**
As per #163 this pull request implements the size, addSelect and member of DQL functions.

### CODE REVIEW

**Comment by peter-gribanov** [comment]
Good work :+1: , but please separate this PR to several feature PRs.

File: src/Spec.php [hunk 1]
```diff
@@ -351,6 +364,31 @@ public static function instanceOfX($value, $dqlAlias = null)
...
+    /**
+     * @param string $field
+     * @param string $value
+     * @param string $dqlAlias
+     *
+     * @return Comparison
+     */
+    public static function memberOf($field, $value, $dqlAlias = null)
+    {
+        return new MemberOf($field, $value, $dqlAlias = null);
```
[the remark this answers is not in the document]
Comment (peter-gribanov) [reply src/Spec.php]: Although, i think it is better to make a separate specification, like [`INSTANCE OF`](https://github.com/Happyr/Doctrine-Specification/blob/master/src/Filter/InstanceOfX.php).

**Comment by patrickvale** [comment]
Hi and thanks for the feedback. Sorry it's taken me a while to get back about it.

Regarding separating the features out into separate PRs do you mean one for each new function eg size(), addSelect() etc?

I did start making Member Of into a MemberOfX specification, like InstanceOfX, but I changed course into making it just a function - I'll did into my code to see why I did that - I think I may have not wanted to create too much boilerplate without checking first.

Likewise with size(), I'll go back to the code that uses it and refresh myself on the use case I needed it for.

Regarding addSelect() could you expand on how you use getQueryBuilder() to modify the result and get the joined fields added into the initial/original query?

**Comment by peter-gribanov** [comment]
> Regarding separating the features out into separate PRs do you mean one for each new function eg size(), addSelect() etc?

Yes. I meant it.

> Regarding addSelect() could you expand on how you use getQueryBuilder() to modify the result and get the joined fields added into the initial/original query?

For example, select only one field:

```php
$my_spec = // ...
$qb = $rep->getQueryBuilder($my_spec, 'u');
$qb->select('u.id');
```

Add entity to result:

```php
$my_spec = Spec::andX(
    Spec::join('author', 'u'),
    // ...
);
$qb = $rep->getQueryBuilder($my_spec);
$qb->addSelect('u');
```

With regard to the `size()` and  `addSelect()` specification. I have an [idea](https://github.com/Happyr/Doctrine-Specification/issues/169) how best to implement it. Please see and say your opinion.

**Comment by peter-gribanov** [comment]
Now you can using `select` and `addSelect` in specification:
```php
$spec = Spec::andX(
    Spec::select('field_name1'),
    Spec::addSelect('field_name2')
)
```

**Comment by peter-gribanov** [comment]
Closed in favor #186

**Pull request closed without merging**